File: Code/new_fitervari_backend/src/main/java/at/htl/model/Trainer.java
package at.htl.model;


import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.time.LocalDate;

@Entity
@Table(name = "trainer")
public class Trainer extends Person {

    @Column(name = "trainer_since")
    private LocalDate trainerSince;

    //region GetterSetter
    public LocalDate getTrainerSince() {
        return trainerSince;
    }

    public void setTrainerSince(LocalDate trainerSince) {
        this.trainerSince = trainerSince;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Trainer.kt
package at.htl_leonding.model

import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "trainer")
class Trainer : Person() {

    @Column(name = "trainer_since")
    lateinit var trainerSince: LocalDate

    fun copyValues(other: Trainer) {
        this.pictureId = other.pictureId
        this.trainerSince = other.trainerSince
    }
}